caddyhttp/staticfiles/fileserver.go

```diff
@@ -121,13 +125,41 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request, name stri
 		return http.StatusNotFound, nil
 	}
 
+	filename := d.Name()
+
+	for _, encoding := range staticEncodingPriority {
+		if strings.Contains(r.Header.Get("Accept-Encoding"), encoding) {
+			gf, err := fs.Root.Open(location + staticEncoding[encoding])
+
+			if err == nil {
+				stat, err := gf.Stat()
+
+				if err == nil {
+					// Close previous file - release fd
+					f.Close()
```

Comment:
@wendigo Did you mean [this](https://github.com/golang/go/blob/master/src/os/file_unix.go#L143) line? That would generate a EINVAL error. Otherwise I wouldn't know how the `*File` pointer would be nil.

Well either way it's nearly a no-op. In that case I'm sorry for "accusing" you. I know that my opinion doesn't "really" count since I'm neither maintainer nor contributor of this PR, but in the end I'd still say that the critique of the double-close is valid -  on one hand because defer is on the heavy side of inbuilt operations and on the other because it appears to be cleanlier in my opinion if you reduce the amount of duplicate (apparent) work. Kinda like how you use bounded channels in Go instead of unbounded ones.